pkg/trace/api/api.go

```diff
@@ -445,7 +446,7 @@ func decodeTracerPayload(v Version, req *http.Request, ts *info.TagStats) (tp *p
 		return &pb.TracerPayload{
 			LanguageName:    ts.Lang,
 			LanguageVersion: ts.LangVersion,
-			ContainerID:     req.Header.Get(headerContainerID),
+			ContainerID:     getContainerID(req),
```

Comment:
I think we need to consider applying this change to `pkg/trace/api/otlp.go` too. The fact that one can omit this makes me wonder where the bad design decision was. Ultimately it would've been ideal if you could've just made this change in one place and that would've applied to the OTLP receiver too...

Reply:
I think the issue is we've duplicated the logic for getting the container id from a request here and [in otlp.go](https://github.com/DataDog/datadog-agent/blob/b9768bc86ca2ba9182b0d9cf89a1d73a4a554886/pkg/trace/api/otlp.go#L245-L251)

Oddly, the logic looks slightly different...

We could (should?) have had a `getContainerID(*http.Request)` function from the beginning that they shared so that the logic could be updated in one place. I'll work on moving this to otlp as well.

Reply:
SGTM!

Reply:
👍 Yes, I agree.

Reply:
👍 Yes, I agree.